components/knowledge/Notion.tsx

```diff
@@ -0,0 +1,204 @@
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  Table,
+  TableBody,
+  TableCell,
+  TableColumn,
+  TableHeader,
+  TableRow,
+} from '@nextui-org/react';
+import React, { useContext, useEffect, useState } from 'react';
+import { IoMdRefresh } from 'react-icons/io';
+import { Image } from '@nextui-org/image';
+import {
+  getNotionFiles,
+  isNotionConfigured,
+  runNotionSync,
+} from '@/actions/knowledge/notion';
+import { CiShare1 } from 'react-icons/ci';
+import { EditContext } from '@/contexts/edit';
+import { importFiles } from '@/actions/knowledge/filehelper';
+
+interface NotionFileModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  isSyncing: boolean;
+  setIsSyncing: React.Dispatch<React.SetStateAction<boolean>>;
+  notionConfigured: boolean;
+  setNotionConfigured: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const NotionFileModal = ({
+  isOpen,
+  onClose,
+  notionConfigured,
+  setNotionConfigured,
+  isSyncing,
+  setIsSyncing,
+}: NotionFileModalProps) => {
+  const { droppedFiles, setDroppedFiles } = useContext(EditContext);
+
+  const [notionFiles, setNotionFiles] = useState<
+    Map<string, { url: string; fileName: string }>
+  >(new Map());
+
+  const [syncError, setSyncError] = useState('');
+  const [selectedFile, setSelectedFile] = useState<string[]>(
+    Array.from(droppedFiles.keys())
+  );
+
+  useEffect(() => {
+    if (notionConfigured) return;
+
+    const setConfigured = async () => {
+      setNotionConfigured(await isNotionConfigured());
+    };
+    setConfigured();
+  }, [notionConfigured, isOpen]);
+
+  useEffect(() => {
+    if (!notionConfigured) return;
+
+    const setFiles = async () => {
+      const notionFiles = await getNotionFiles();
+      setNotionFiles(notionFiles);
+    };
+
+    setFiles();
+  }, [isOpen, notionConfigured]);
+
+  const onClickImport = async () => {
+    const files = await importFiles(selectedFile, 'notion');
+    setDroppedFiles((prev) => {
+      const newMap = new Map(prev);
+      for (const file of Array.from(newMap.entries())) {
```

Comment:
nit: same here. Destructuring is easier to understand
```suggestion
      for (const [fileName, fileDetail] of Array.from(newMap.entries())) {
```